next-app/app/matches/page.js

```diff
@@ -68,25 +78,18 @@ function MatchData({ matches }) {
   );
 }
 
-// Formats the data by condensing each match to 1 line, adding the players and their respective teams, and removing unnecessary information (pname and team).
-function formatData(data) {
-  let formatted = [];
-  let players = { teamA: [], teamB: [] };
-  for (let i = 0; i < data.length; i++) {
-    const { pname, team, ...rest } = data[i];
-    if (team === 'a') {
-      players.teamA.push(pname);
-    } else {
-      players.teamB.push(pname);
-    }
-
-    // run this right before match cutover (on the last item in data for a match)
-    if (i === data.length - 1 || data[i].match_id !== data[i + 1].match_id) {
-      Object.assign(rest, players);
-      rest.mdate = new Date(rest.mdate);
-      formatted.push(rest);
-      players = { teamA: [], teamB: [] };
-    }
-  }
-  return formatted;
+function reformatMatches(matches) {
+  const reformat = matches.map((value) => {
+    const { teamA, teamB, ...rest } = value;
+    const a = value.teamA.map((a) => {
+      return a.pname;
+    });
+    const b = value.teamB.map((b) => {
+      return b.pname;
+    });
+    rest.teamA = a;
+    rest.teamB = b;
```

Comment:
you could skip a step and set the properties directly (also, remove the `value.` before accessing the team since you already destructured it out)
```suggestion
    rest.teamA = teamA.map((a) => {
      return a.pname;
    });
    rest.teamB = teamB.map((b) => {
      return b.pname;
    });
```